cpp/tests/utilities/type_lists.hpp

```diff
@@ -64,21 +64,35 @@ constexpr auto types_to_ids()
 
 }  // namespace detail
 
+/**
+ * @brief Convert numeric values type T to numeric vector of type TypeParam.
+ *
+ * This will also convert negative values to positive values if the output type is unsigned.
+ *
+ * @param init_list Values used to create the output vector
+ * @return Vector of TypeParam with the values specified
+ */
 template <typename TypeParam, typename T>
 auto make_type_param_vector(std::initializer_list<T> const& init_list)
 {
   std::vector<TypeParam> vec(init_list.size());
   std::transform(std::cbegin(init_list), std::cend(init_list), std::begin(vec), [](auto const& e) {
-    return static_cast<T>(e);
+    if (std::is_unsigned<TypeParam>::value) return static_cast<TypeParam>(std::abs(e));
+    return static_cast<TypeParam>(e);
   });
   return vec;
 }
 
+/**
+ * @brief Type list for all integral types except type bool.
+ */
+using IntegralTypesNotBool =
```

Comment:
What's the purpose of this?

Reply:
There were a couple of places where we were recreating the entire list of integer types just to mask out the bool type. That would be hard to find in the future if we add new types. So I thought that should go here.

Reply:
_Should_ the bool type be excluded though? Seems like that's a good way to forget to add tests for bool.

Reply:
It's just a convenience. The `IntegralsTypes` includes bool as it should.
For example, there are some places in libcudf we handle integers and but throw an exception for bool: [partition](https://github.com/rapidsai/cudf/blob/403d2571e5fcde66b7768b2213f6c142cc8b63db/cpp/src/partitioning/partitioning.cu#L711-L721), [gather](https://github.com/rapidsai/cudf/blob/403d2571e5fcde66b7768b2213f6c142cc8b63db/cpp/src/copying/gather.cu#L61-L68), [scatter](https://github.com/rapidsai/cudf/blob/403d2571e5fcde66b7768b2213f6c142cc8b63db/cpp/src/copying/scatter.cu#L61-L73)

Also, it is nice to be able to write a `TYPED_TEST` for all the integers and then a separate test for checking the bool-exception or handling bool differently.

Reply:
The gather/scatter examples are a good point. Makes sense to have this.